api/python/provisioner/commands/__init__.py

```diff
@@ -592,14 +622,20 @@ def run(self, targets):  # noqa: C901 FIXME
                     # TODO: Improve salt minion restart logic
                     # please refer to task EOS-14114.
                     try:
+                        logger.info("Restarting salt minions.")
                         _restart_salt_minions()
                     except Exception:
-                        logger.exception('failed to restart salt minions')
+                        logger.exception(
+                           "FAILED: Restarting salt minions. "
```

Comment:
Don't you think that `ERROR` label for such message is not enough and we need `FAILED` or something like that?
If we really need that then there is an alternative option (I guess not only one actually).

Example: configure logger to add additional field for such additional labels and pass the as extra data for a log record.

From [docs](https://docs.python.org/3.6/library/logging.html#logging.debug):

>The third optional keyword argument is extra which can be used to pass a dictionary which is used to populate the __dict__ of the LogRecord created for the logging event with user-defined attributes. These custom attributes can then be used as you like. For example, they could be incorporated into logged messages

@nitin-seagate I believe it is similar what we discussed about recently: optional sub-levels for the standard log levels.

Reply:
> keyword argument is extra which can be used to pass a dictionary which is used to populate the **dict** of the LogRecord created for the logging event with user-defined attributes. These custom attributes can then be used as you like. For example, they could be incorporated into logged messages

Overall I would support this, but should be a separate PR. This one's scope should be limited to some deliverable.

Reply:
@divya-vijay please create one more separate follow-up task for that as well